src/main/webapp/app/entities/study/study-detail.component.ts

```diff
@@ -25,11 +28,26 @@ export class StudyDetailComponent implements OnInit, OnDestroy {
     ) {
     }
 
+    onButtonClicked(currentStatus:boolean){
+        if (this.presetSend === currentStatus){
+            this.presetSend = !this.presetSend;
+        }
+    }
+
     ngOnInit() {
         this.subscription = this.route.params.subscribe((params) => {
             this.load(params['id']);
         });
         this.registerChangeInStudies();
+        this.currentTimeStamp = StudyService.getCurrentDateTime();
+        this.currentTimeZone = this.getTimeZone();
+    }
+
+    // credit for function: https://stackoverflow.com/questions/1091372/getting-the-clients-timezone-in-javascript
+    // User: Mr_Green
+    getTimeZone(): string {
+        var offset = new Date().getTimezoneOffset(), o = Math.abs(offset);
+        return (offset < 0 ? "+" : "-") + ("00" + Math.floor(o / 60)).slice(-2) + ":" + ("00" + (o % 60)).slice(-2);
```

Comment:
That's quite a long return line?
I will store that in a return variable personally

Reply:
hmm, then the variable assignment would be a long line?
I got that function from SO, which why I didn't modify it